Speedup display of library text and fixes

chrjorgensen opened the pull request:
### Changes

This PR will
- speed up the library list view by omitting multiple (unnecessary) calls to get library information
- fix error when library was current library and also in library list (it would not show in the list part, only as current library)
- fix error displaying `(current library)` (when library was also in library list, `(current library)` would show in both places)

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [x] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [ ] **for feature PRs**: PR only includes one feature enhancement.

worksofliam reviewed (changes requested):
@chrjorgensen Seems to be a little bug when the library list has an entry when the library does not exist - and when this happens the User Library List view does not render anymore. Perhaps if this happens, we still should display the entry in the list and either: a) show no description or b) the description mentions that it was not found

<img width="530" alt="image" src="https://user-images.githubusercontent.com/3708366/191087885-b6a2cde1-53dd-4cd9-8897-2c55eadf592b.png">

chrjorgensen commented:
@worksofliam Good catch! 👍 

I have now changed `getLibraryList` to always return all the libraries in the list - if a library does not exist, it will have the text
`*** NOT FOUND ***` (commit [0b76ead](https://github.com/halcyon-tech/vscode-ibmi/pull/858/commits/0b76eade50b9acecd68269ac01bf8442555a365f))

I have also added a button for refresh of the library list and a command and button to validate the library list:

![billede](https://user-images.githubusercontent.com/13275072/191116299-ae8fef91-1be5-4cd1-8c4f-56e093a0b7c4.png)

Now the user can more easily get the current descriptions of the libraries in the list and validate / remove any libraries no longer on the server.

worksofliam commented:
Neat, that is a nice change.

I will get a review in ASAP this week!

Also, do you think that refresh and validation should be one button? I'm okay to have them as two buttons, but wonder if the UX is practically the same. Any ideas?

chrjorgensen commented:
A refresh will show the user if any libraries have been deleted or renamed - without removing them from the library list.
A validate will remove any libraries that have been deleted or renamed - and the library list will be changed and the user can't keep his/her list, if the library delete/rename was a mistake.

I understand your reasoning, though - one could argue that validate and refresh should be one operation. However, currently I think it should be up to the user to decide what to do with the library list: Either refresh and get a warning or sanitize the list and remove any libraries no longer available.

If screen space is an issue, the validate button could be moved to the secondary actions area in the view? The refresh button should always be available, like it is in the other views,

chrjorgensen commented:
@worksofliam @Wright4i 
I agree on your thoughts and have renamed the function `Validate library list` to `Cleanup library list` to inform the user, that the library list will be cleaned and may change.

I also changed the button to a secondary action, which now shows in the `...` - together with the variable maintenance button.
I think the change gives a cleaner look for the library list view:

![billede](https://user-images.githubusercontent.com/13275072/191586959-4568c1a0-1ad0-4b6a-9ea5-b3325a6c21de.png)

What do you think?

Wright4i commented:
Works for me!
